### PR TITLE
Upgrade to latest terracotta-platform and terracotta-core versions.

### PR DESCRIPTION
**Pull request by chrisdennis**

### CODE REVIEW

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ChainCodec.java [hunk 1]
```diff
@@ -26,12 +26,24 @@
...
+import org.terracotta.runnel.encoding.StructEncoderFunction;
 
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.List;
 
-public class ChainCodec {
+public final class ChainCodec {
+
+  private ChainCodec() {
```
Comment (chrisdennis) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ChainCodec.java]: I made this class static... was there some reason we wanted instances of it?

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ExceptionCodec.java [hunk 2]
```diff
@@ -26,11 +26,23 @@
...
 import org.terracotta.runnel.encoding.StructEncoder;
+import org.terracotta.runnel.encoding.StructEncoderFunction;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
 
-class ExceptionCodec {
+final class ExceptionCodec {
+
+  private ExceptionCodec() {
```
Comment (chrisdennis) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ExceptionCodec.java]: I made this one static too... was that wrong?

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ResponseCodec.java [hunk 3]
```diff
@@ -91,24 +89,13 @@
     .byteBuffer(MAP_VALUE_FIELD, 20)
     .build();
 
-  private final ChainCodec chainCodec;
-
-  public ResponseCodec() {
```
Comment (chrisdennis) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ResponseCodec.java]: I might be able to make this one static now too... but I didn't.  Should I have done?

File: build.gradle [hunk 4]
```diff
@@ -50,13 +50,13 @@ ext {
   sizeofVersion = '0.3.0'
 
   // Clustered
-  terracottaPlatformVersion = '5.2.0-pre1'
+  terracottaPlatformVersion = '5.2.0-pre2'
```
Comment (mathieucarbou) [inline build.gradle]: should be `5.2.0-pre3` (I've released again after a bugfix)

**Comment by mathieucarbou** [comment]
@chrisdennis : I approve the PR, though tc-platform version should be changed to __5.2.0-pre3__. Otherwise there will be issue with M&M when integrating into "other" projects.

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ChainCodec.java (hunk 1 above)
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ChainCodec.java]: None I can think of

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ExceptionCodec.java (hunk 2 above)
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ExceptionCodec.java]: Don't think so

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ResponseCodec.java (hunk 3 above)
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ResponseCodec.java]: Maybe, can happen at a later stage though.

**Pull request merged**